erpnext/accounts/doctype/pos_invoice_merge_log/pos_invoice_merge_log.py

```diff
@@ -83,6 +87,11 @@ def on_submit(self):
 		pos_invoice_docs = [
 			frappe.get_cached_doc("POS Invoice", d.pos_invoice) for d in self.pos_invoices
 		]
+		accounting_dimensions = get_accounting_dimensions()
```

Comment:
Why set in every POS invoice, this can be set just in the final consolidated invoice

Reply:
So , if there is any return, then 2 consolidated invoices will be created, so will have to add it twice anyway.